src/SoftDeleteable/Query/TreeWalker/SoftDeleteableWalker.php

```diff
@@ -89,8 +132,8 @@ private function getSoftDeleteableListener()
         if (null === $this->listener) {
             $em = $this->getEntityManager();
 
-            foreach ($em->getEventManager()->getListeners() as $event => $listeners) {
-                foreach ($listeners as $hash => $listener) {
+            foreach ($em->getEventManager()->getListeners() as $listeners) {
+                foreach ($listeners as $listener) {
                     if ($listener instanceof SoftDeleteableListener) {
                         $this->listener = $listener;
 
```

Comment:
Couldn't we use `break 2` in the next line and remove the extra `break` condition in the first `foreach`?

Reply:
Probably.  But, I'd do this change as a separate PR because there are a bunch of classes with a very similar loop.